src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -326,14 +326,32 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
 
     DWORD ConptyConnection::_OutputThread()
     {
-        UTF8OutPipeReader pipeReader{ _outPipe.get() };
-        std::string_view strView{};
+        std::array<char, 4096> buffer{}; // buffer for the chunk read.
+        UTF8ChunkToUTF16Converter convertUTF8ChunkToUTF16{};
+        std::wstring_view u16Sv{};
 
         // process the data of the output pipe in a loop
         while (true)
         {
-            HRESULT result = pipeReader.Read(strView);
-            if (FAILED(result) || result == S_FALSE)
+            DWORD read{};
+
+            const auto readFail{ !ReadFile(_outPipe.get(), &buffer.at(0), gsl::narrow<DWORD>(buffer.size()), &read, nullptr) };
```

Comment:
Done.

FWIW I recall suggesting just this in #2607 but finally you used the & operator. Do you remember if using .data() was an issue in the static analysis?

Reply:
I don't recall if the static analysis is why. I don't think it is since you changed it and the gate passed. Also, that PR was very big. So I might have made a few mistakes like this.

Reply:
OK, thanks.